packages/core-api/src/apis/implementations/auth/helpers.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ConfigApi } from '../../definitions';
+
+export function getDefaultScopesFromConfig(
+  configApi: ConfigApi,
+  { provider, environment }: { provider: string; environment: string },
+): Set<string> {
+  const defaultScopes = configApi.getOptionalStringArray(
+    `auth.providers.${provider}.${environment}.defaultScopes`,
+  );
+  return new Set(defaultScopes);
```

Comment:
Thinking that this affects users in an upgrade path, seen as they wont have the `defaultScopes` part in their yaml. Could we default to `read:user` here if there are none in the config for now for at least a few releases?

Reply:
This definitely makes sense! Should we start with just the GitHub provider or should we implement it for everyone? I created that PR to wait for the general feedback.

Reply:
I think just github provider for now.

Reply:
Changed it to @Rugvip's suggestion. If there is a need to change the other providers as wll. Let me know! Ready to contribute 😄